tictactoe-game/script.js

```diff
@@ -1,50 +1,94 @@
-var player = 1;
-var moves = 0;
-
-function play(square) {
-	if(square.innerHTML==='') {
-		moves++;
-
-		if(player===1) {
-			square.innerHTML = 'X';
-			player = 2;
-		}
-		else {
-			square.innerHTML = 'O';
-			player = 1;
-		}
-		
-		var result = checkGameOver();
-		
-		if( result!==false ) {
-			document.getElementById('message').innerHTML = 'Game Over: '+result;
-		}
-		else {
-			document.getElementById('message').innerHTML = 'Keep playing';
-		}
-	}
-}
-
-function checkGameOver() {
-	// 1st: check if theres less 5 moves
-	if(moves<5) {
-		return false;
-	}
-	
-	// 2nd: check for winner lines
-	var row1 = document.getElementsByClassName('r1');
-	
-	if( row1[0].innerHTML!=='' &&
-			row1[0].innerHTML===row1[1].innerHTML &&
-			row1[0].innerHTML===row1[2].innerHTML) {
-		return row1[0].innerHTML + ' is the winner.';
-	}
-	
-	// 3rd: check if board is full
-	if(moves>8){
-		return "It's a tie";
-	}
-	// continue game
-	else
-		return false;
-}
\ No newline at end of file
+var player = 1;
+var moves = 0;
+var gameFinish = false;
+function play(square) {
+	if ((square.innerHTML === '') && (!gameFinish)) {
+		moves++;
+		if(player===1) {
+			square.innerHTML = 'X';
+			player = 2;
+		}else {
+			square.innerHTML = 'O';
+			player = 1;
+		};
+		var result = checkGameOver();
+		if( result!==false ) {
+			gameFinish = true;
+			document.getElementById('message').innerHTML = 'Game Over: '+result;
+		}else {
+			document.getElementById('message').innerHTML = 'Keep playing';
+		};
+	};
+};
+function checkGameOver() {
+	// 1st: check if theres less 5 moves
+	if(moves<5) {
+		return false;
+	};
+	// 2nd: check for winner lines
+	var result = winner();
+	if (result) {
+		return result;
+	};
+	// 3rd: check if board is full
+	if(moves>8){
+		return "It's a tie";
+	}// continue game
+	else {
+		return false;
+	};
+};
+function winner() {
+	var arr = document.getElementsByClassName('sq');
+	var i = 0;
+	var j = 0;
+	if ((arr[0].innerHTML !== '') && (arr[0].innerHTML === arr[4].innerHTML) && (arr[0].innerHTML === arr[8].innerHTML)) {

```

Comment:
I would use either classes or `data-*` updates to keep track of this sort of thing. It's a much more precise way of handling it, as any character would cause a win.

I don't know for certain, but it probably has much better performance too, so it's a win for clarity and a win for performance (however slight).